test/unit/dashboard_test.rb

```diff
@@ -0,0 +1,88 @@
+require 'test_helper'
+
+class DashboardTest < ActiveSupport::TestCase
+  setup do
+    @host = FactoryGirl.create(:host, :with_reports)
+    @env = FactoryGirl.create(:environment)
+  end
+
+  test 'hosts returns correct host' do
+    data = Dashboard::Data.new
+    as_admin do
+      assert_equal 1, data.hosts.length
+    end
+  end
+
+  test 'hosts works with environment filter' do
+    @host.update_attribute(:environment_id, @env.id)
```

Comment:
Pass `:environment => @env` to the host creation in `setup` to avoid these additional updates in many of the tests.